austation/code/modules/antagonists/bloodsucker/bloodsucker_life.dm

```diff
@@ -0,0 +1,536 @@
+			return TRUE
+
+		// Limbs? (And I have no other healing)
+		if (amInCoffinWhileTorpor)
+
+			// Heal Missing
+			var/list/missing = owner.current.get_missing_limbs()
+			if (missing.len)
+				// Cycle through ALL limbs and regen them!
+				for (var/targetLimbZone in missing)
+					// 1) Find ONE Limb and regenerate it.
+					//var/targetLimbZone = pick(missing)
+					owner.current.regenerate_limb(targetLimbZone, 0)		// regenerate_limbs() <--- If you want to EXCLUDE certain parts, do it like this ----> regenerate_limbs(0, list("head"))
+					// 2) Limb returns Damaged
+					var/obj/item/bodypart/L = owner.current.get_bodypart( targetLimbZone )
+					AddBloodVolume(20 * costMult)	// Costs blood to heal
+					L.brute_dam = 60
+					to_chat(owner.current, "<span class='notice'>Your flesh knits as it regrows [L]!</span>")
+					playsound(owner.current, 'sound/magic/demon_consume.ogg', 50, 1)
+
+
+				// DONE! After regenerating ANY number of limbs, we stop here.
+				return TRUE
+
+			/*else // REMOVED: For now, let's just leave prosthetics on. Maybe you WANT to be a robovamp.
+				// Remove Prosthetic/False Limb
+				for(var/obj/item/bodypart/BP in C.bodyparts)
+					message_admins("T1: [BP] ")
+					if (istype(BP) && BP.status == 2)
+						message_admins("T2: [BP] ")
+						BP.drop_limb()
+						return TRUE
+						// NOTE: Limbs have a "status", like their hosts "stat". 2 is dead (aka Prosthetic). 1 seems to be idle/alive.
+			*/
+
+			// Cure Final Disabilities
+			CureDisabilities()
+
+			// Remove Embedded!
+			C.remove_all_embedded_objects()
+			// Heal Organs (will respawn original eyes etc. but we replace right away, next)
+			owner.current.regenerate_organs()
+			// Eyes/Heart
+			CheckVampOrgans() // Heart, Eyes
+
+	return FALSE
+
+
+/datum/antagonist/bloodsucker/proc/CureDisabilities()
+	var/mob/living/carbon/C = owner.current
+
+	C.cure_blind(list(EYE_DAMAGE))//()
+	C.cure_nearsighted(EYE_DAMAGE)
+	C.set_blindness(0) 	// Added 9/2/19
+	C.set_blurriness(0) // Added 9/2/19
+	C.update_tint() 	// Added 9/2/19
+	C.update_sight() 	// Added 9/2/19
+
+	for(var/O in C.internal_organs) //owner.current.adjust_eye_damage(-100)  // This was removed by TG
+		var/obj/item/organ/organ = O
+		organ.setOrganDamage(0)
+
+	owner.current.cure_husk()
+
+
+// I am hungry!
+/datum/antagonist/bloodsucker/proc/HandleStarving()
+
+	// High: 	Faster Healing
+	// Med: 	Pale
+	// Low: 	Twitch
+	// V.Low:   Blur Vision
+	// EMPTY:	Frenzy!
+
+	// BLOOD_VOLUME_GOOD: [336]  Pale (handled in bloodsucker_integration.dm
+
+
+	// BLOOD_VOLUME_BAD: [224]  Jitter
+	if (owner.current.blood_volume < BLOOD_VOLUME_BAD && !prob(0.5))
+		owner.current.Jitter(10)
+
+	// BLOOD_VOLUME_SURVIVE: [122]  Blur Vision
+	if (owner.current.blood_volume < BLOOD_VOLUME_BAD / 2)
+		owner.current.blur_eyes(8 - 8 * (owner.current.blood_volume / BLOOD_VOLUME_BAD))
+
+	// Nutrition
+	owner.current.nutrition = min(owner.current.blood_volume, NUTRITION_LEVEL_FED) // <-- 350  //NUTRITION_LEVEL_FULL
```

Comment:
This is causing mood swings, because nutrition level will try to fall from NUTRITION_LEVEL_FED (and gets reset to NUTRITION_LEVEL_FED every time HandleStarving() is called) and corresponding positive mood_event will flicker when blood_volume is >350.

Reply:
Known about it since I first played blood suckers. I might make a different noodlet that had a longer timeout then it flickers...

Reply:
I have an idea, do the same thing with food or something and remove bloodsuckers from HandleNutrition, have separate moodlets for having blood and having low blood. That removes the whole ticking issue as nutrition is constantly fighting over the moodlets.

Reply:
This is a very minor graphical issue and can still be testmerged but would need to have a look at this before fully merging this.

Reply:
```suggestion
	// Nutrition
	owner.current.nutrition = min(owner.current.blood_volume, NUTRITION_LEVEL_FED) // <-- 350  //NUTRITION_LEVEL_FULL
```
```suggestion
	// Nutrition
	owner.current.nutrition = min(owner.current.blood_volume, NUTRITION_LEVEL_FED - 1) // <-- 349  //NUTRITION_LEVEL_FULL - 1 // Removes nutrition mood flickering
```

This should work if you just want to remove the flickering; exempting bloodsucker from HandleNutrition can't be modularized, sadly.